test/e2e/custom_wait_rules_test.go

```diff
@@ -222,34 +273,38 @@ metadata:
 		}
 		patch := `[{ "op": "replace", "path": "/status/currentState", "value": "Running"}]`
 		PatchClusterResource("CronTab", "my-new-cron-object-1", env.Namespace, patch, kubectl)
-	}()
+	}
 
 	logger.Section("deploy resource with current state as progressing", func() {
 		kapp.RunWithOpts([]string{"deploy", "-f", "-", "-a", crdApp}, RunOpts{StdinReader: strings.NewReader(crd)})
+		go patchConfigMap()
 
-		out, _ := kapp.RunWithOpts([]string{"deploy", "-f", "-", "-a", name}, RunOpts{
-			StdinReader: strings.NewReader(yaml)})
+		kapp.RunWithOpts([]string{"deploy", "-f", "-", "-a", name}, RunOpts{StdinReader: strings.NewReader(cr)})
+
+		patchCustomResource()
+		//remove age column from output, as we can't rely on it
+		columns := "namespace,name,kind,op,'op strategy','wait to','reconcile state','reconcile info'"
+		out, _ := kapp.RunWithOpts([]string{"deploy", "-f", "-", "-a", name, "--column", columns}, RunOpts{
+			StdinReader: strings.NewReader(yaml + cr)})
 
 		out = strings.TrimSpace(replaceSpaces(replaceTarget(replaceTs(out))))
 
 		expectedOutput := strings.TrimSpace(replaceSpaces(`Changes
 
-Namespace  Name                  Kind       Age  Op      Op st.  Wait to    Rs  Ri  $
-kapp-test  my-new-cron-object-1  CronTab    -    create  -       reconcile  -   -  $
-^          test-cm               ConfigMap  -    create  -       reconcile  -   -  $
+Namespace  Name                  Kind       Op      Op st.  Wait to    Rs  Ri  $
+kapp-test  my-new-cron-object-1  CronTab    -       -       reconcile  ok  -  $
+^          test-cm               ConfigMap  create  -       reconcile  -   -  $
 
-Op:      2 create, 0 delete, 0 update, 0 noop, 0 exists
+Op:      1 create, 0 delete, 0 update, 1 noop, 0 exists
 Wait to: 2 reconcile, 0 delete, 0 noop
 
-<replaced>: ---- applying 1 changes [0/2 done] ----
-<replaced>: create crontab/my-new-cron-object-1 (stable.example.com/v1) namespace: kapp-test
-<replaced>: ---- waiting on 1 changes [0/2 done] ----
-<replaced>: ongoing: reconcile crontab/my-new-cron-object-1 (stable.example.com/v1) namespace: kapp-test
-<replaced>:  ^ Allowing blocked changes to proceed: Unblock blocked changes
-<replaced>: ---- applying 1 changes [1/2 done] ----
+<replaced>: ---- applying 2 changes [0/2 done] ----
```

Comment:
We want to test that when the status reaches Progressing for the CR, we apply the ConfigMap, but due to #729 that doesn't happen, the ConfigMap gets applied before that.

Reply:
I just realised that we can use default values for these fields in the CRD definition instead of patching them initially. Something like this:
```yaml
            status:
              type: object
              default: {}
              properties:
                currentState:
                  type: string
                  default: Progressing
```

Reply:
^ We can do this in other tests as well.

Reply:
Yesh, I think we are good 🚀